### DISCUSSION TITLE
A2DP source with aptx encoding

### DISCUSSION THREAD

**Discussion by Hasti-Gondaliya** [pschatzmann/arduino-audio-tools#1833] [Q&A] [answered]
Hi everyone,

I'm currently working on a project where I aim to develop an A2DP source that takes headphone-level audio and transmits it to a Bluetooth sink using the ESP32-Wroover module. Additionally, I'd like to add support for APTX to improve the audio quality.

However, I'm encountering some challenges with the following:
1. Setting up the A2DP source correctly to ensure reliable audio transmission.
2. Configuring and implementing APTX support with the ESP32-Wroover.

I'm fairly new to these hardware components and Bluetooth audio protocols, so any guidance, code snippets, or resources would be greatly appreciated.

**Comment by pschatzmann** [comment, marked as answer]
To my knowledge the Espressif API does not support this!

**Reply by Hasti-Gondaliya** [reply]
Thanks a lot for your quick response @pschatzmann.
Is it feasible to develop this with Arduino? I've reviewed some examples from ESP32-A2DP and audio-tools, but I'm uncertain if I can accomplish my goal using those.

**Reply by Hasti-Gondaliya** [reply]
Hello @pschatzmann ,

Thanks again for your quick response. I'm wondering if there are any alternative approaches or additional resources that might help me develop an A2DP source with APTX support using the ESP32-Wroover. Any guidance would be greatly appreciated.

**Comment by pschatzmann** [comment]
Submit a feature request to Espressif or use a A2DP hardware module
